mlx/backend/metal/compiled.cpp

```diff
@@ -329,7 +329,9 @@ void Compiled::eval_gpu(
       if (in.flags().row_contiguous && in.nbytes() == outputs[o].nbytes() &&
           in.is_donatable() &&
           constant_ids_.find(inputs_[i].id()) == constant_ids_.end()) {
-        outputs[o++].move_shared_buffer(in);
+        outputs[o].move_shared_buffer(
+            in, outputs[o].strides(), in.flags(), in.data_size());
```

Comment:
Just to verify my understanding. The only way `output[o].strides()` is not the same as `in.strides()` is if the output has a bunch of unsqueezed dimensions in the beginning right? Otherwise the input would not be `row_contiguous` or `in.nbytes() != outputs[o].nbytes()`.

Reply:
Yea that's right. I'm pretty sure it's the only case.